### ISSUE TITLE
global window properties are not loaded when loading an external script

### ISSUE THREAD

**Issue by yinonby** [codesandbox/codesandbox-client#8534]
# 🐛 bug report

## Description of the problem
When loading external script, some global window properties should be added.
Specifically, after loading 'mocha' and 'chai' scripts, the following properties should exist: 'window.mocha' and 'window.chai'.
This works well in a normal environment.
However, when running the same code in a codesandbox, these properties are missing from the global 'window' variable.

## How has this issue affected you? What are you trying to accomplish?
My goal is to create a codesandbox to demonstrate Mocha capability of running in a browser, as specified here:
https://mochajs.org/#running-mocha-in-the-browser

### To Reproduce
Create a Node environment with a simple html, sourcing the following script:
> <script src="https://unpkg.com/chai/chai.js"></script>

Then, in another script, try to access `window.chai`

In a non-codesandbox environment, you will get a structure.
In codesandbox, you will get `undefined`.

So, when trying to access `window.chai.expect`, the following Type Error occurs:
> Cannot read properties of undefined (reading 'expect')

The reason is that `chai` is undefined in `window`

### Link to sandbox:
[link to sandbox](https://codesandbox.io/s/mocha-chai-browser-test-p3l9c?file=/index.html)

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Сodesandbox      |        Latest      |
| Browser          |        Chrome Version 127.0.6533.88 (Official Build) (arm64)      |
| Operating System |       MacOS Sonoma 14.2.1 (23C71)       |

**Comment by CompuIves** [comment]
Hey! Yes, the entry point (index.html) is determined based on the `main` field in `package.json`. In this case, the `main` field is `index.html`, so it expects to have an entry point of `index.html` inside the root. I _think_ that's the behaviour, but I have to admit that due to how Parcel is implemented it might also show inconsistent behaviour with order of loading scripts. Our bundler parses the `index.html`, and then "emits" transpiled files and executes them based on the script tags.

In this case, that should work though. You mention that putting the `index.html` in the root solves the problem, but I cannot repro that. Could it be that I'm missing something?

**Comment by yinonby** [comment]
Good point, I reviewed it, and indeed placing the html in the project root, in the sandbox I provided, doesn't solve the issue.
It did solve it in my own repository though (another project).

So I isolated it further (in the sandbox I provided), and it seems the `<script>` tags are simply not running.
Take a look at this [sandbox](https://codesandbox.io/p/sandbox/html-script-test-d7q7sp?file=%2Findex1.html)
It is a simplified project, with practically a basic html, with 2 `<div>` components, and a `<script>` in between, which outputs to the console.
You will notice that nothing is being output.

Does this make any sense?

**Comment by CompuIves** [comment]
Ah! That makes sense, there's a discrepancy with our bundler resolving and the one that our static server does. If you open https://d7q7sp.csb.app/index1.html, then you should see that the script tags _do_ work.

The parcel template is a bit hacky, I think that's why we see this behaviour. In this case, the frontend bundler (the one running in the browser) resolves the `index1.html` because it knows from `package.json` it needs to resolve this. It does `document.write` with the contents of the `index1.html`,  but that means that the script tags are not executed.

Meanwhile, our static server does _not_ check `package.json` to know what file to resolve. Because of this, the file is not served. If the file would be renamed to `index.html`, then both our bundler and static server would resolve to the same file, meaning that the script tags get executed.

Does that make sense? There are some intricacies from how this was implemented sprinkled around here 😅 .
